### DISCUSSION TITLE
Scalability of graph spaces

### DISCUSSION THREAD

**Discussion by Sahil333** [vesoft-inc/nebula#5983] [Q&A]
Hi All,

We have a project in which we are thinking to use nebula graph database to represent the file system of different cloud systems of our clients. Initially, we have thought to use one graph space for one cloud storage with 1 partition count as we think meta information of a file system will be able to store on a single machine. A client can have many cloud storages - say 100. And we want to scale to about 1000 clients. Will nebula graph be able to scale graph spaces to 100,000?  The reason for this decision to keep the latency low for traversal queries so that query doesn't need to span multiple nodes.

I do see in the documentation that it uses multi-group raft in storaged layer that share resources like in transport layer and thread pool. But still I would like to know some expert's opinion before taking this decision.

The other option is to setup graph spaces at the client level and keeping a higher partition count than 1.

**Comment by xiajingchun** [comment]
100,000 space sounds crazy to me, though there's no hard limit on that. If I understand correctly, even considering the data privacy among you clients, at least you can create one space for each client, that would be 1000 spaces, still a lot but much more reasonable now.
another thing, even you want to use only 1 node and 1 cloud storage, it's still better to have a reasonable partition number than 1, storaged can query partition data in parallel. 

May I know you data volume, both in terms of GB/TB and the number of nodes/edges?
